pac4j-core/src/main/java/org/pac4j/core/credentials/authenticator/LocalCachingAuthenticator.java

```diff
@@ -56,8 +59,30 @@ public void validate(final T credentials, final WebContext context) throws HttpA
             });
             credentials.setUserProfile(profile);
             logger.debug("Found cached credential. Using cached profile {}...", profile);
-        } catch (Exception e) {
-            throw new CredentialsException(e);
+        } catch (UncheckedExecutionException e) {
+            if (e.getCause() instanceof RuntimeException) {
+                throw (RuntimeException) e.getCause();
+            } else if (e.getCause() != null) {
```

Comment:
The only reason I would argue against or pro the else in that place is the following: without the else, maintainability is threatened. It is easier to introduce bugs by modifying this code if there is no explicit structure about what happens and when. It is the same thing as when removing the braces around the argument of a `if`: there is much more chance to add a statement that the developer think will be executed as part of the `if` while actually it won't.
Now, I really think there is no reason to spend more time discussing that, so I will simply follow your recommendations and be done with it :)